### ISSUE TITLE
Down migrations didn't execute on rollback

### ISSUE THREAD

**Issue by jpascal** [uptrace/bun#99]
I have this migrations: 
```
20210812164558_uuid.up.sql
20210812164558_uuid.down.sql
20210812170337_users.down.sql
20210812170337_users.up.sql
```
On ```migrator.Migrate(ctx)```:
```
2021-08-13T18:00:23.720+0300    WARN    database/logger.go:54   INSERT INTO schema_migrations_lock ("id", "table_name") VALUES (DEFAULT, 'schema_migrations') RETURNING "id"    {"env": "dev", "elapsed": "350.4372ms", "source": "database", "operation": "INSERT"}
2021-08-13T18:00:23.849+0300    DEBUG   database/logger.go:50   SELECT * FROM schema_migrations {"env": "dev", "elapsed": "108.7393ms", "source": "database", "operation": "SELECT"}
2021-08-13T18:00:23.911+0300    DEBUG   database/logger.go:50
CREATE EXTENSION "uuid-ossp";
        {"env": "dev", "elapsed": "61.962ms", "source": "database", "operation": "\nCREATE"}
2021-08-13T18:00:24.028+0300    DEBUG   database/logger.go:50   INSERT INTO schema_migrations ("id", "name", "group_id", "migrated_at") VALUES (DEFAULT, '20210812164558', 1, DEFAULT) RETURNING "id", "migrated_at"    {"env": "dev", "elapsed": "117.1954ms", "source": "database", "operation": "INSERT"}
2021-08-13T18:00:24.878+0300    WARN    database/logger.go:54   CREATE TABLE users
(
    id  uuid default uuid_generate_v4() NOT NULL CONSTRAINT users_pkey PRIMARY KEY,
    created_at       TIMESTAMP WITH TIME ZONE NOT NULL,
    updated_at       TIMESTAMP WITH TIME ZONE NOT NULL,
    admin            BOOLEAN DEFAULT FALSE NOT NULL,
    active           BOOLEAN DEFAULT FALSE NOT NULL,
    email            TEXT UNIQUE NOT NULL,
    hashed_password VARCHAR(100)
);
        {"env": "dev", "elapsed": "850.3139ms", "source": "database", "operation": "CREATE"}
2021-08-13T18:00:25.062+0300    DEBUG   database/logger.go:50   INSERT INTO schema_migrations ("id", "name", "group_id", "migrated_at") VALUES (DEFAULT, '20210812170337', 1, DEFAULT) RETURNING "id", "migrated_at"    {"env": "dev", "elapsed": "183.3398ms", "source": "database", "operation": "INSERT"}
2021-08-13T18:00:25.213+0300    DEBUG   database/logger.go:50   DELETE FROM schema_migrations_lock WHERE ("table_name" = 'schema_migrations')   {"env": "dev", "elapsed": "151.5975ms", "source": "database", "operation": "DELETE"}
2021-08-13T18:00:25.213+0300    INFO    migrate/main.go:125     migrated to group #1 (20210812164558, 20210812170337)
        {"env": "dev"}
```

On ```migrator.Rollback(ctx)```

```
2021-08-13T18:00:37.752+0300    WARN    database/logger.go:54   INSERT INTO schema_migrations_lock ("id", "table_name") VALUES (DEFAULT, 'schema_migrations') RETURNING "id"    {"env": "dev", "elapsed": "421.3177ms", "source": "database", "operation": "INSERT"}
2021-08-13T18:00:37.886+0300    DEBUG   database/logger.go:50   SELECT * FROM schema_migrations {"env": "dev", "elapsed": "109.8567ms", "source": "database", "operation": "SELECT"}
2021-08-13T18:00:37.944+0300    DEBUG   database/logger.go:50   DELETE FROM schema_migrations WHERE (id = 3)    {"env": "dev", "elapsed": "58.1305ms", "source": "database", "operation": "DELETE"}
2021-08-13T18:00:38.001+0300    DEBUG   database/logger.go:50   DELETE FROM schema_migrations WHERE (id = 4)    {"env": "dev", "elapsed": "57.2276ms", "source": "database", "operation": "DELETE"}
2021-08-13T18:00:38.060+0300    DEBUG   database/logger.go:50   DELETE FROM schema_migrations_lock WHERE ("table_name" = 'schema_migrations')   {"env": "dev", "elapsed": "58.757ms", "source": "database", "operation": "DELETE"}
2021-08-13T18:00:38.060+0300    INFO    migrate/main.go:141     rolled back group #1 (20210812164558, 20210812170337)
        {"env": "dev"}
```

Why down migrations didn't executed?

**Comment by jdbann** [comment]
Yeah, I'm getting the same issue.

It looks like `migrate.MigrationsWithStatus` replaces the registered `Migration`s with the ones loaded from the DB and so don't have the `Up` and `Down` functions: https://github.com/uptrace/bun/blob/master/migrate/migrator.go#L70-L75

**Comment by vmihailenco** [comment]
Thanks for the report - this should be fixed in v0.4.1

**Comment by jdbann** [comment]
Ah great! Wanted to have a crack at fixing it but wasn't sure without tests in the area - thanks for adding those in too @vmihailenco!

**Comment by vmihailenco** [comment]
You are welcome :+1:

**Comment by jdbann** [comment]
Ah no, I'm sorry to say this @vmihailenco but the rollback doesn't run the migrations in reverse order...

**Comment by vmihailenco** [comment]
@jdbann should be fixed in v0.4.2

**Comment by jdbann** [comment]
Perfect. Thank you!
